app/controllers/api/v1/locations_controller.rb

```diff
@@ -27,9 +27,15 @@ def create
   end
 
   def destroy
-    Location.find(params["id"]).destroy
-    locations = Location.all
-    render json: locations
+    location = Location.find(params["id"])
+
+    if current_user.id == location.user_id
+      Location.find(params["id"]).destroy
+      locations = Location.all
+      render json: locations
```

Comment:
since you're not doing anything with this data in your fetch (correct me if I'm wrong), you'll want to do something like 
```
render json: { message: "Location was deleted" }
```

Even if you don't end up displaying this message, it could be helpful for debugging, and it doesn't make an unnecessary database call or confuse other devs :)